src/resources/js/parts-kit.js

```diff
@@ -181,6 +133,141 @@ class Toggle {
     };
 }
 
-// Init the toggle class
-const toggles = document.querySelectorAll("[data-parts-kit-toggle]");
-toggles.forEach((el) => new Toggle(el));
+// Parts kit class for expanding/collapsing sidebar
+class PartsKit {
+    constructor(el) {
+        this.el = el;
+        this.setVars();
+        this.bindEvents();
+        this.initSidebar();
+    }
+
+    setVars() {
+        this.sidebar = this.el.querySelector("[data-sidebar]");
+        this.lsSidebar = "pk-sidebar-hidden";
+        this.fullscreenExpandBtn = this.el.querySelector(
+            "[data-fullscreen-expand]"
+        );
+        this.fullscreenCloseBtn = this.el.querySelector(
+            "[data-fullscreen-close]"
+        );
+
+        // Set up toggle classes
+        const toggleElements = document.querySelectorAll("[data-parts-kit-toggle]");
+        this.toggles = [...toggleElements].map(el => new Toggle(el));
+
+        /** @type {HTMLInputElement} */
+        this.sidebarSearch = this.el.querySelector("[data-sidebar-search]");
+        /** @type {NodeListOf<HTMLElement>} */
+        this.sectionItems = this.el.querySelectorAll("[data-section-item]");
+        this.searchList = [...this.sectionItems].map(el => {
+            const toggleEl = el.closest("[data-parts-kit-toggle]");
+            return {
+                el,
+                searchText: el.getAttribute('data-search-text')?.toLowerCase(),
+                toggleEl,
+                toggle: this.toggles.find(x => x.el === toggleEl)
+            }
+        });
+    }
+
+    bindEvents() {
+        document.addEventListener("keydown", this.handleKeydown);
+        this.fullscreenExpandBtn.addEventListener("click", this.toggleSidebar);
+        this.fullscreenCloseBtn.addEventListener("click", this.toggleSidebar);
+        this.sidebarSearch.addEventListener("input", this.handleSearchInput);
+        this.sidebarSearch.addEventListener("focus", ({target}) => {
+            target.setAttribute('data-blur-placeholder', target.getAttribute('placeholder'))
+            target.setAttribute('placeholder', target.getAttribute('data-focus-placeholder'))
+        })
+        this.sidebarSearch.addEventListener("blur", ({target}) => {
+            target.setAttribute('placeholder', target.getAttribute('data-blur-placeholder'))
+        })
+    }
+
+    noUserInput = () => document.activeElement.tagName !== 'INPUT'
+
+    handleKeydown = ({ keyCode }) => {
```

Comment:
keyCode is deprecated... https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode

I think using [key](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key) or [code](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code) is that new new